cogs/embeds.py

```diff
@@ -1,24 +1,142 @@
+                    current = ""
+                
+                if current is not None:
+                    current = current + token + " "
+                
+                if token.endswith("\"") and current is not None:
+                    index = len(token) - 2
+                    if index < 0 or token[index] != "\\":
+                        data = current.strip()
+                        data = data[1:-1]
+                        output.append(data)
+                        current = None
+
+            return output
+
+        async def handle_channel():
+            while True:
+                try:
+                    channel_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+                    if len(channel_msg.channel_mentions) is 0:
+                        await ctx.send("Could not find that channel")
+                    else:
+                        channel = channel_msg.channel_mentions[0]
+                        return channel
+                except TimeoutError:
+                    await ctx.send("Reached Timeout. Discarding Embed")
+                    break
+
+        async def handle_replacement_msg(embed: discord.Embed, channel: discord.TextChannel):
+            msg_id_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+            msg_id = msg_id_msg.content.strip()
+            try:
+                msg = await channel.fetch_message(msg_id)
+                await msg.edit(embed=embed)
+            except:
+                await channel.send(embed=embed)
+
+        timeout = 60 * 1000
+        embed = discord.Embed()
+        
+        await ctx.send("Please type the description of the embed. Use `skip` to skip this.")
+        desc: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+        if not is_skip(desc): embed.description = desc.content.strip()[:4096]
+
+        await ctx.send("Please provide the colour in hex format (e.g. `#123456`). Use `skip` to skip this.")
+        await handle_color(embed)
+
+        await ctx.send("Please provide the URL for the image you would like to display. Use `skip` to skip this.")
+        await handle_img(embed)
+
+        await handle_fields(embed)
+        
+        await ctx.send("Which channel should the embed be sent in?")
+        channel = await handle_channel()
+
+        await ctx.send("Should this embed replace any existing message in that channel? If yes, type the message ID, otherwise type anything else.")
+        await handle_replacement_msg(embed, channel)
+
+
     @commands.command()
-    async def embedraw(self, ctx: commands.Context):
+    async def embedraw(self, ctx: commands.Context, *, embed_data):
         """
         Create a Discord embed via raw JSON input.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
+        try:
+            embed_body = json.loads(embed_data)
```

Comment:
This implementation works, but it is not how `embedraw` works within IB.ai - the JSON used there takes the `fields` values as a list of 2-lists (a title string and a value string). However, I'm not sure if we want to preserve backwards compatibility or change this - maybe wait for feedback from Ray or Nathaneal.

Reply:
I think maintaining backwards compatibility would be good. Since I recall a lot of the embed_raw commands are pre-written and it'd be annoying to have to edit all of those.

Reply:
Can you provide me a sample snippet of the JSON that we are currently using for testing?

Reply:
Also, do we handle inline fields in the old method, and if not, should we be adding that functionality? It is a matter of parsing the JSON and checking whether the array has 2 elements or 3.